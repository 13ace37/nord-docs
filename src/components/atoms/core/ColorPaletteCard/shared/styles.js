/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides multiple styles for the `ColorPaletteCard` component.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.6.0
 */

import { rgba } from "polished";

import { colors, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const floatingLineColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.nord4, 0.8),
  [MODE_DARK_NIGHT_FROST]: rgba(colors.nord4, 0.5)
});

const dropShadowColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.shadow.base[MODE_BRIGHT_SNOW_FLURRY], 0.35),
  [MODE_DARK_NIGHT_FROST]: rgba(colors.shadow.base[MODE_DARK_NIGHT_FROST], 0.2)
});

const fontColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord3,
  [MODE_DARK_NIGHT_FROST]: colors.nord4
});

export { dropShadowColor, floatingLineColor, fontColor };
