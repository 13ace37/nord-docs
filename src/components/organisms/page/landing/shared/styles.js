/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared styles for the landing page section components.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.6.0
 */

import { css } from "styled-components";
import { rgba } from "polished";

import {
  colors,
  motion,
  themedMode,
  zIndexFor,
  MODE_BRIGHT_SNOW_FLURRY,
  MODE_DARK_NIGHT_FROST,
  Z_INDEX_ELEMENTS
} from "styles/theme";

const darkenedThemeModeBackgroundbackgroundColor = themedMode({
  [MODE_DARK_NIGHT_FROST]: colors.background.sectioning.tertiary[MODE_DARK_NIGHT_FROST]
});

const decorationBaseStyles = css`
  position: absolute;
  z-index: ${zIndexFor(Z_INDEX_ELEMENTS.DECORATIONS)};
  transition: fill ${motion.speed.duration.transition.base.themeModeSwitch}ms ease-in-out;
`;

const decorationSectionColorPaletteModularityFillColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord5,
  [MODE_DARK_NIGHT_FROST]: colors.nord1
});

const decorationSectionHeroFillColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.nord6, 0.4),
  [MODE_DARK_NIGHT_FROST]: colors.nord1
});

const decorationSectionColorSwatchFillColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord6,
  [MODE_DARK_NIGHT_FROST]: colors.nord2
});

export {
  darkenedThemeModeBackgroundbackgroundColor,
  decorationBaseStyles,
  decorationSectionColorPaletteModularityFillColor,
  decorationSectionColorSwatchFillColor,
  decorationSectionHeroFillColor
};
