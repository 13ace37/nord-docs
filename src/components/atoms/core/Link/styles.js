/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides multiple styles for the `Link` component.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.5.0
 */

import { css } from "styled-components";
import { lighten, rgba } from "polished";

import { colors, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const backgroundColorHover = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.nord6, 0.45),
  [MODE_DARK_NIGHT_FROST]: rgba(colors.nord3, 0.8)
});

const fontColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord10,
  [MODE_DARK_NIGHT_FROST]: colors.nord8
});

const fontColorDecent = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord0,
  [MODE_DARK_NIGHT_FROST]: lighten(0.2, colors.font.base[MODE_DARK_NIGHT_FROST])
});

const fontColorHover = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord10
});

const fontColorHoverMinimal = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord0,
  [MODE_DARK_NIGHT_FROST]: lighten(0.2, colors.font.base[MODE_DARK_NIGHT_FROST])
});

const calmly = css`
  border-radius: 0.25em;
  color: ${fontColor};
  transition: color 0.2s ease-in-out, background-color 0.2s ease-in-out;

  &:hover,
  &:active,
  &:focus {
    background-color: ${backgroundColorHover};
    color: ${fontColorHover};
  }
`;

const decent = css`
  border-bottom: 1px solid ${fontColor};
  color: ${fontColorDecent};
  transition: color 0.2s ease-in-out;

  &:hover,
  &:active,
  &:focus {
    color: ${fontColor};
  }
`;

const minimal = css`
  border-bottom: 1px solid ${fontColor};
  transition: border-bottom-color 0.2s ease-in-out;

  &:hover,
  &:active,
  &:focus {
    border-bottom-color: ${fontColorHoverMinimal};
  }
`;

export { calmly, decent, minimal };
