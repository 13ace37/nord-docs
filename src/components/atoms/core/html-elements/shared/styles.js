/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared component styles.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.10.0
 */

import { colors, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const borderColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord6,
  [MODE_DARK_NIGHT_FROST]: colors.nord2
});

/* eslint-disable-next-line import/prefer-default-export */
export { borderColor };
