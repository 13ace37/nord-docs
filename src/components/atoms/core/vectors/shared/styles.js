/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared vector graphic component styles.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */

import { css } from "styled-components";

import { colors, motion, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const themeModeFillColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.font.base[MODE_BRIGHT_SNOW_FLURRY],
  [MODE_DARK_NIGHT_FROST]: colors.font.base[MODE_DARK_NIGHT_FROST]
});

/**
 * The value and transition for the fill color based on the active global theme mode.
 *
 * @since 0.3.0
 */
const themeModeFillColorStyles = css`
  fill: ${themeModeFillColor};
  transition: fill ${motion.speed.duration.transition.base.themeModeSwitch}s ease-in-out;
`;

/* eslint-disable-next-line import/prefer-default-export */
export { themeModeFillColorStyles };
