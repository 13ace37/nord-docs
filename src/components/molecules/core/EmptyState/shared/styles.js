/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared component styles.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */

import { css } from "styled-components";
import { darken } from "polished";

import { colors, motion, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const svgFillColorOutlines = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.font.base[MODE_BRIGHT_SNOW_FLURRY],
  [MODE_DARK_NIGHT_FROST]: darken(0.045, colors.nord0)
});

const svgFillColorSnowflakes = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.font.base[MODE_BRIGHT_SNOW_FLURRY],
  [MODE_DARK_NIGHT_FROST]: colors.nord4
});

const svgTransitionFill = css`
  transition: fill ${motion.speed.duration.transition.base.themeModeSwitch}ms ease-in-out;
`;

export { svgFillColorOutlines, svgFillColorSnowflakes, svgTransitionFill };
