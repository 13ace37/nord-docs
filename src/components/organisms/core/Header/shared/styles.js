/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared component styles.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */

import { rgba } from "polished";

import { colors, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

/**
 * The factor with which the branding logo is multiplied by the height of the header based on the pin mode.
 *
 * @type {number}
 */
const HEADER_BRAND_LOGO_SHRINK_FACTOR = 0.45;

/**
 * The default height (in `px`) of the header when in unpinned mode.
 *
 * @type {number}
 */
const HEADER_HEIGHT = 80;

/**
 * The default height (in `px`) of the header for the pinned mode in pixels.
 *
 * @type {number}
 */
const HEADER_HEIGHT_PINNED = 56;

const backgroundColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.background.base[MODE_BRIGHT_SNOW_FLURRY],
  [MODE_DARK_NIGHT_FROST]: colors.background.base[MODE_DARK_NIGHT_FROST]
});

const linkBackgroundColorHover = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.nord6, 0.4),
  [MODE_DARK_NIGHT_FROST]: rgba(colors.nord3, 0.8)
});

export {
  backgroundColor,
  linkBackgroundColorHover,
  HEADER_BRAND_LOGO_SHRINK_FACTOR,
  HEADER_HEIGHT,
  HEADER_HEIGHT_PINNED
};
