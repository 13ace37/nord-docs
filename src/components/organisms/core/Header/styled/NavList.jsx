/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import { ms } from "styles/theme";

/**
 * A flexbox list for navigation links with dynamic spacing.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const NavList = styled.div`
  display: none;
  justify-content: space-between;
  font-size: ${ms(1)};

  * {
    margin: 0 0.556em;

    &:last-child {
      margin-right: 0;
    }

    &:first-child {
      margin-left: 0;
    }
  }

  ${({ theme }) => theme.media.tabletPortrait`
    display: flex;
  `};
`;

export default NavList;
