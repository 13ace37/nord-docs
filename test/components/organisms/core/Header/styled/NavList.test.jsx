/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";
import { em } from "polished";

import { renderWithTheme } from "nord-docs-test-utils";
import NavList from "organisms/core/Header/styled/NavList";
import { media } from "styles/theme";

describe("theme styles", () => {
  test("has required CSS breakpoints", () => {
    const { container } = renderWithTheme(<NavList />);
    const mediaQuery = `(min-width: ${em(media.breakpoints.tabletPortraitLowerBoundary)})`;

    expect(container.firstChild).toHaveStyleRule("display", "none");
    expect(container.firstChild).toHaveStyleRule("display", "flex", {
      media: mediaQuery
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
