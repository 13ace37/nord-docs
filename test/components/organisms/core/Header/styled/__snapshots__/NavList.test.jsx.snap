// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme styles has required CSS breakpoints 1`] = `
.c0 {
  display: none;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 1.125em;
}

.c0 * {
  margin: 0 0.556em;
}

.c0 *:last-child {
  margin-right: 0;
}

.c0 *:first-child {
  margin-left: 0;
}

@media (min-width:37.5em) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  class="c0"
/>
`;
